content/4_Gold/LIS.mdx

```diff
@@ -105,10 +105,12 @@ these are $[2]$, $[2, 5]$, and $[2, 5, 15]$.
 **Lemma 1:** $L_i$ forms a strictly increasing sequence.
 
 **Proof:** Assume for a contradiction that for some $j$, we have
-$L_i[j] \geq L_i[j + 1]$. Define $a_i$ ($0 \leq i \leq j$) to be the $i$-th
-element of any increasing subsequence that ends with $L_i[j + 1]$. We can have
-an increasing subsequence of length $j$ that ends on $a_{j - 1}$, which is less
-than $a_j = L_i[j + 1]$, so there is a contradiction.
+$L_i[j] \geq L_i[j + 1]$. Let $a_0, a_1, \dots, a_j$ be any increasing subsequence
+of length $j+1$ ending on $L_i[j + 1]$ (so $a_j=L_i[j + 1]$). Notice that
+$a_0, a_1, \dots, a_{j-1}$ is an increasing subsequence of length $j$.
+Thus, $L_i[j]$ cannot be greater than $a_{j-1}$.
+From that, we can write this equation: $L_i[j] \leq a_{j-1} < a_j = L_i[j+1]$,
+which is exactly what we wanted.
```

Comment:
Uhh, is this still accurate per the new definition of L_i[j]?

I believe indicies should be shifted by 1

Reply:
Oh yeah, I'll change that.